src/cmds/pbs_rsub.c

```diff
@@ -256,6 +258,13 @@ process_opts(int argc, char **argv, struct attrl **attrp, char *dest)
 				}
 				break;
 
+			case '-':
+				if (strcmp(optarg, "hosts") == 0)
+					is_maintenance_resv = 1;
```

Comment:
Will this impact all vnodes assigned to the list of hosts?

Reply:
The vnodes are obtained later with pbs_statvnode(). For each host passed as a parameter (here), the associated vnodes are found and used in crafting execvnodes.